src/llvm-version.h

```diff
@@ -8,8 +8,8 @@
 #define JL_LLVM_VERSION (LLVM_VERSION_MAJOR * 10000 + LLVM_VERSION_MINOR * 100 \
                         + LLVM_VERSION_PATCH)
 
-#if JL_LLVM_VERSION != 30300 && JL_LLVM_VERSION < 30701
-    #error LLVM versions < 3.3 are not supported by Julia
+#if JL_LLVM_VERSION < 30600
```

Comment:
At least that's what we currently support (although LLVM 3.3 doesn't pass all tests anymore IIRC). We'll probably drop 3.3 support and require at least LLVM 3.7.1 (or even LLVM 3.8?) during the 0.6 cycle but that should probably be another PR (since this PR is already pretty big as it is now.)